src/acp/util/getAgentAccessAll.test.ts

```diff
@@ -0,0 +1,613 @@
+/**
+ * Copyright 2022 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { ACL, ACP } from "../constants";
+import {
+  DEFAULT_ACCESS_CONTROL_RESOURCE_URL,
+  DEFAULT_DOMAIN,
+  TEST_URL,
+} from "../mock/constants";
+import { createDatasetFromSubjects } from "../mock/dataset";
+import { mockAccessControlledResource } from "../mock/mockAccessControlledResource";
+import { getAgentAccessAll } from "./getAgentAccessAll";
+
+describe("getAgentAccessAll", () => {
+  it("returns an empty object for an empty ACR", async () => {
+    const resource = mockAccessControlledResource();
+
+    expect(resource.internal_acp.acr.graphs).toStrictEqual({ default: {} });
+    expect(await getAgentAccessAll(resource)).toStrictEqual({});
+  });
+
+  it("returns the access modes for a simple ACR", async () => {
```

Comment:
I think it would help if the test description was more precise: what is a "simple ACR" in this context ? I assume an ACR with a single policy.

Reply:
Yes ^^,
Or more precisely, one policy, one condition, one matcher, no denying...

Maybe an ACR that uses the minimal structure?
I'm not sure how to convey it really accurately.

Reply:
`returns the access modes for a simple ACR (ne policy, one condition, one matcher, no denying)` seems like a good start. If that's too long, maybe `returns the access modes for a minimal ACR`, with "minimal" described in a comment in the test body would work too.